src/garage/tf/baselines/gaussian_cnn_baseline.py

```diff
@@ -199,9 +198,15 @@ def __init__(self,
         self._initialize()
 
     def _initialize(self):
-        input_var = tf.compat.v1.placeholder(tf.float32,
-                                             shape=(None, ) +
-                                             self._input_shape)
+        if isinstance(self._env_spec.observation_space, akro.Image):
+            input_var = tf.compat.v1.placeholder(tf.uint8,
+                                                 shape=(None, ) +
+                                                 self._input_shape)
+            input_var = tf.cast(input_var, tf.float32) / 255.0
```

Comment:
this defeats the purpose of using a uint8 placeholder -- you want only uint8 to feedforward through the network

Reply:
Got it.

Reply:
@ryanjulian we can't normalize pixel values to between 0 and 1 if we don't cast to a float though. AFAIK using uint8 as the input placeholder dtype optimizes memory usage before data is passed to the CNN, not while feeding forward?

Reply:
Right, uint8 works in replay buffer where no normalization is required.

Reply:
if observation_space is not Image, uint8 could still save memory so we should use uint8 by default, and only cast to float and normalize if observation_space is Image?

Reply:
The primitive then wouldn't support float data out of the box though. We could just add the dtype as a param in the constructor with uint8 as the default.